### PR TITLE
Add parameters for threaded solver and accumulator.

### PR DESCRIPTION
**Pull request by gbalduzz**
Quick PR to add a couple of options.

### CODE REVIEW

**Comment by uhaehner** [comment]
test this please

File: include/dca/phys/parameters/mci_parameters.hpp [hunk 1]
```diff
@@ -86,7 +96,9 @@ class MciParameters {
   int measurements_;
   int walkers_;
   int accumulators_;
+  bool shared_walk_and_accumulation_thread_;
   bool adjust_self_energy_for_double_counting_;
+  int error_computation_type_;
```
Comment (biddisco) [inline include/dca/phys/parameters/mci_parameters.hpp]: why is this an int and not an ErrorComputationType

File: include/dca/phys/parameters/mci_parameters.hpp [hunk 2]
```diff
@@ -67,9 +71,15 @@ class MciParameters {
...
     return accumulators_;
   }
+  bool shared_walk_and_accumulation_thread() const {
+    return shared_walk_and_accumulation_thread_;
+  }
   bool adjust_self_energy_for_double_counting() const {
     return adjust_self_energy_for_double_counting_;
   }
+  ErrorComputationType get_error_computation_type() const {
+    return ErrorComputationType(error_computation_type_);
```
Comment (biddisco) [inline include/dca/phys/parameters/mci_parameters.hpp]: A converion would not be necessary if this was ErrorComputationType

File: include/dca/phys/parameters/mci_parameters.hpp [hunk 3]
```diff
@@ -177,15 +195,25 @@ void MciParameters::readWrite(ReaderOrWriter& reader_or_writer) {
...
     }
+
     try {
       reader_or_writer.execute("measurements", measurements_);
     }
     catch (const std::exception& r_e) {
     }
 
+    // Read error computation type.
+    std::string error_type = toString(static_cast<ErrorComputationType>(error_computation_type_));
```
Comment (biddisco) [inline include/dca/phys/parameters/mci_parameters.hpp]: same here. No real need for a type cast

File: include/dca/phys/parameters/mci_parameters.hpp (hunk 1 above)
Comment (gbalduzz) [reply include/dca/phys/parameters/mci_parameters.hpp]: I would need to add specializations to MPITypeMap for each enumeration. Do you think it is a better idea or would there be a way to specialize the class in a way to match all possible enumerations?

**Comment by biddisco** [comment]
OK. I spent a bit of time experimenting with this and the existing implementation is much more straightforward than using a scoped enumerated type in place of an int in the member function. The simple int/enum conversion is less intrusive than having to add an MPI pack overload. Switching to a simple enumerated type with implicit conversion to int suffers from name collision with FourPointType::NONE and cleaning it all up would be a new PR/Issue.

**Pull request merged**